collector/template.bpf.c

```diff
@@ -46,8 +46,8 @@ int uprobe_compass_php_function(struct pt_regs *ctx) {
   bpf_probe_read_user_str(&event->request_id, STRSZ, (void *)ctx->PHP_FUNCTION_ARG_REQUEST_ID);
   bpf_probe_read_user_str(&event->class_name, STRSZ, (void *)ctx->PHP_FUNCTION_ARG_CLASS_NAME);
   bpf_probe_read_user_str(&event->function_name, STRSZ, (void *)ctx->PHP_FUNCTION_ARG_FUNCTION_NAME);
-  event->start_time = ctx->PHP_FUNCTION_ARG_START_TIME;
-  event->end_time = ctx->PHP_FUNCTION_ARG_END_TIME;
+  event->timestamp = bpf_ktime_get_ns();
+  event->elapsed = ctx->PHP_FUNCTION_ARG_ELAPSED;
```

Comment:
That is actually clever.

Because you are now no longer affecting performance you can get the ts in the event instead of from the actual function in PHP. Event ordering is still giving the parent hierarchy, right?

Another way - though slightly more effort is to create a real realtime clock TS at start of request.

And then pass this realtime start, monotonic start and monotonic end always to the probe! - that way it’s both realtime and not really.

Reply:
It has been working well. I really like decoupling the timestamps and letting the probe me a metric.